user-script

```diff
@@ -72,9 +95,15 @@ case "$mode" in
 	'') fatal 'mode must be specified as the first argument';;
 	*) fatal "invalid mode specified: '$mode'";;
 esac
-[[ -x $script ]] || fatal "script $script not found or executable"
+[[ -x $script ]] || fatal "script $script not found or not executable"
 
 # run the script
-msg "running $script for user $user (DISPLAY=:$display)"
-sudo -Hu "$user" "DISPLAY=:$display" timeout "$timeout" "$script"
+if [[ $display == *"wayland"* ]]; then
+	env="WAYLAND_DISPLAY=$display XDG_RUNTIME_DIR=$xrd"
+else
+	env="DISPLAY=:$display"
+fi
+
+msg "running $script for user $user ($env)"
+sudo -Hu "$user" $env timeout "$timeout" "$script"
```

Comment:
with `env` as an array the expansion will need to become "${env[@]}" for this line:

``` bash
sudo -Hu "$user" "${env[@]}" timeout "$timeout" "$script"
```